app/components/category-item.js

```diff
@@ -15,4 +18,37 @@ export default Ember.Component.extend({
     let userCategories = this.get('userCategories');
     return userCategories.hasCategory(category);
   }),
+
+  actions: {
+    addCategory(category) {
+      this.set('isLoading', true);
+      this.addCategory(category).catch(() => {
```

Comment:
Maybe a separate issue and separate PR. We can come back to it as I'd like to move on to some other things for now.